### DISCUSSION TITLE
Simple glb object do not load in Scene Viewer

### DISCUSSION THREAD

**Discussion by uprealdigital** [google/model-viewer#2951] [Q&A]
### Description

None of the glb objects that I made loads properly in Scene Viewer. It only says "Couldn't load Object". I wish I could see logs or anything else to help me debug this.

#### Live Demo
https://sceneview.glitch.me/

### Version
- model-viewer: v1.9.1

### Browser Affected

- [X] Chrome
- [ ] Edge
- [ ] Firefox
- [X] Internet Samsung
- [ ] IE
- [ ] Safari

### OS

- [X] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR
- [ ] WebXR
- [X] SceneViewer
- [ ] QuickLook

Screenshot:

![Screenshot_20211101-174944_Google](https://user-images.githubusercontent.com/25235922/139747385-a5d9dea9-690b-42e7-bfce-da8c03930a24.jpg)

3D Model:

[3DMenu3_BabylonJS.zip](https://github.com/google/model-viewer/files/7456375/3DMenu3_BabylonJS.zip)

**Comment by elalish** [comment]
Agreed the lack of logs is annoying; that's one of the big disadvantages with intenting out of the browser. With WebXR  you can get a lot more detailed feedback. In this case I'd bet the problem is Glitch, likely something about  that little query param on the end of the GLB's URL. I  tested your model in Scene Viewer using our Editor and  mobile deploy  and it works fine, so there's no problem with your file.

**Reply by valenn** [reply]
Do you know if there is any way at all to tell what ARCore has problem with when it gets you this meaningless message? (isn't both this and ARCore google projects?) It fails to load valid GLB files regularly and in the past I've changed our exporter to have special preprocessing for android (because it e.g. cannot handle 2 objects with same name, even if it is allowed in gltf). But now I am mystified that two meshes that work separately, don't when they are exported together. Using WebXR is not an option because Apple either doesn't want to or cannot implement it, and seeing how long it took them to get to WebGL2 I'm not holding my breath

**Reply by elalish** [reply]
@AdrianAtGoogle regarding Scene Viewer GLB load issues. You already have to use two different AR apps for iOS and Android, since Quick Look is the only option for iOS and not available anywhere else. So the choice of Scene Viewer vs. WebXR is currently a purely Android decision. Why do you consider it not an option?

**Reply by valenn** [reply]
Sorry, I was not talking specifically about model-viewer, this was the only thread with any reply in last few years I could find regarding the "Couldn't load object" error. Regarding WebXR, If you have renderer with e.g. reflections, refractions, subsurface and stuff like this, it might be preferable to go for native AR apps for consistency (they are more or less comparable in looks), at least when it comes to explaining to clients why AR looks different on android/apple. Having the AR on android actually report what error it encountered would be helpful

**Reply by elalish** [reply]
The `<model-viewer>` renderer (three.js) supports more glTF extensions than SceneViewer does, so our WebXR mode is in no way a downgrade. But, if you're not actually using `<model-viewer>` then it doesn't really apply.

**Comment by AdrianAtGoogle** [comment]
So the model in question is _technically_ valid, but it confused our parser; the GLB file has a binary section containing 25844 bytes of data, but _also_ includes said 25k of data as a base64 encoded URI.  The binary section and the base64-decoded data match in size and content.

Given that no other loaders flag this data duplication as an error, I'll get a fix submitted to ignore the situation where the binary content is provided twice.

Per the other question, we don't surface low level loading errors out to the user.  If you have developer mode enabled, you may be able to get logcat output for aGSA, but in this case it wouldn't tell you much, emitting something like `Error: Failed to Load Gltf url 'http://<path>.glb' reason: invalid glTF`
